src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.Linq.cs

```diff
@@ -80,7 +80,9 @@ internal static T[] ToArray<T>(IEnumerable<T> source)
                 return result;
             }
 
-            var builder = new LargeArrayBuilder<T>(initialize: true);
+#pragma warning disable SA1129 // Do not use default value type constructor
```

Comment:
Is this rule still relevant with C# 10's featur to allow parameterless constructors for value types?

Reply:
> Is this rule still relevant with C# 10's featur to allow parameterless constructors for value types?

I think it's even more relevant, but only once the rule is updated to stop warning about use of `new()` with structs that do expose a parameterless ctor.  Then it becomes very meaningful in the code whether you see `new T()` vs `default(T)`.  @sharwell?

Reply:
@sharwell, are there efforts to update SA1129 to account for the new C# features here?

Reply:
Here's the tracking issue:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3430

I should be able to address the bug you hit soon.

Reply:
Looks like this was fixed.

Reply:
Still getting errors https://dev.azure.com/dnceng/public/_build/results?buildId=1604857&view=results
Do I need to update something here?

Reply:
It's updated.  Try again. :)

Reply:
I can pull down the branch and try it later today.

Reply:
> Please give me some hints on how to resolve this.

The problem is you've defined the parameterless ctor in a file that's not part of all System.Linq.dll builds (e.g. LargeArrayBuilder's ctors are defined in LargeArrayBuilder.SpeedOpt.cs).  When it's not built, then SA1129 is warning you that your use of e.g. `new LargeArrayBuilder<T>()` is actually just `default(LargeArrayBuilder<T>)`.

Reply:
Thanks for having a look!
Hope that I got it right in 2ad31865edf6a624f0bb236682d0100b7681084d (local builds are passing, `build.cmd` didn't show that failure, only in VS build which I didn't before).

Reply:
> build.cmd didn't show that failure,

The root build script by default doesn't build every target of every assembly. Building a library in VS or using dotnet build in the library's src directory does.

Reply:
Great. Thanks for that information!